jwst/photom/tests/test_photom.py

```diff
@@ -0,0 +1,1470 @@
+        wl[:] = y.copy()
+    else:
+        raise RuntimeError("dispaxis must be either 1 or 2.")
+
+    return wl
+
+
+def create_input(instrument, detector, exptype,
+                 filter=None, pupil=None, grating=None, band=None):
+    """Create dummy data (an open model) of the appropriate type.
+
+    Parameters
+    ----------
+    instrument : str
+        The instrument name (all upper case letters), one of:
+        'NIRISS', 'NIRSPEC', 'NIRCAM', 'MIRI', 'FGS'.
+
+    detector : str
+        Detector name.  This is only used for populating a keyword.
+
+    exptype : str
+        Exposure type.  These are explicitly checked for:
+        'NIS_WFSS', 'NIS_SOSS', 'NRS_FIXEDSLIT', 'NRS_BRIGHTOBJ',
+        'NRS_MSASPEC', 'NRC_WFSS', 'MIR_MRS', 'MIR_LRS-FIXEDSLIT'.
+
+    filter : str or None
+        Name of the element in the filter wheel.  For NIRISS WFSS, this
+        is used to determine the dispersion direction.
+
+    pupil : str or None
+        Name of the element in the pupil wheel.  For NIRCam WFSS, this
+        is used to determine the dispersion direction.
+
+    grating : str or None
+        Name of the element in the grating wheel.  This is only used for
+        populating a keyword.
+
+    band : str or None
+        Band (MIRI only).  This is only used for populating a keyword.
+
+    Returns
+    -------
+    input_model : `~jwst.datamodels.DataModel`
+        An open data model object of the appropriate type.
+    """
+
+    if instrument == 'NIRISS':
+        if exptype == 'NIS_WFSS':
+            nslits = 2
+            input_model = datamodels.MultiSlitModel()
+            if filter.endswith('R'):
+                shape = (69, 5)
+                dispaxis = 2                    # vertical
+            else:
+                shape = (5, 69)
+                dispaxis = 1                    # horizontal
+            (data, dq, err, var_p, var_r, var_f) = mk_data(shape)
+            wl = mk_wavelength(shape, 1.0, 5.0, dispaxis)
+            for k in range(nslits):
+                slit = datamodels.SlitModel(data=data, dq=dq, err=err,
+                                            wavelength=wl)
+                slit.var_poisson = var_p
+                slit.var_rnoise = var_r
+                slit.var_flat = var_f
+                slit.meta.wcsinfo.spectral_order = k + 1
+                input_model.slits.append(slit.copy())
+        elif exptype == 'NIS_SOSS':
+            shape = (96, 2048)
+            (data, dq, err, var_p, var_r, var_f) = mk_data(shape)
+            input_model = datamodels.ImageModel(data=data, dq=dq, err=err)
+            # There is no wavelength attribute for ImageModel, but this
+            # should work anyway.
+            wl = mk_wavelength(shape, 0.6, 4.0, dispaxis=1)
+            input_model.wavelength = wl.copy()
+            input_model.var_poisson = var_p
+            input_model.var_rnoise = var_r
+            input_model.var_flat = var_f
+        else:
```

Comment:
What exptype does this last branch get used for? Or is it just a catch-all for any random exptype?

Reply:
I added a comment to say that this is for NIS_IMAGE.